quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -581,18 +562,6 @@ void index(BuildProducer<IndexDependencyBuildItem> indexDependencyBuildItemBuild
         indexDependencyBuildItemBuildProducer.produce(new IndexDependencyBuildItem("org.keycloak", "keycloak-services"));
     }
 
-    @BuildStep(onlyIf = IsJpaStoreEnabled.class, onlyIfNot = IsLegacyStoreEnabled.class)
-    @Consume(CheckJdbcBuildStep.class)
-    void indexNewJpaStore(BuildProducer<IndexDependencyBuildItem> indexDependencyBuildItemBuildProducer) {
-        indexDependencyBuildItemBuildProducer.produce(new IndexDependencyBuildItem("org.keycloak", "keycloak-model-map-jpa"));
-    }
-
-    @BuildStep(onlyIf = IsLegacyStoreEnabled.class)
-    @Consume(CheckJdbcBuildStep.class)
-    void indexLegacyJpaStore(BuildProducer<IndexDependencyBuildItem> indexDependencyBuildItemBuildProducer) {
-        indexDependencyBuildItemBuildProducer.produce(new IndexDependencyBuildItem("org.keycloak", "keycloak-model-jpa"));
-    }
```

Comment:
This should not be completely removed. Only the condition in the `@BuildStep` annotation and probably rename the method to `indexJpaStore`.